apps/common/behaviours/casework-submission.js

```diff
@@ -6,10 +6,9 @@ const client = new StatsD();
 
 const Compose = func => superclass => class extends superclass {
   prepare() {
-    if (typeof func === 'function') {
-      return Object.assign(super.prepare(), func(this.toJSON()));
-    }
-    return super.prepare();
+    return this.auth().then(token => {
```

Comment:
This prepare step essentially overrides the one that exists in the HOF model. Superclass declared above is essentially either the `apps/common/models/i-casework.js` or `apps/common/models/i-casework-documents.js`; the latter which extends off the former anyway. The i-casework model has now been updated to extend off the Auth model and HOF models allowing us to call this.auth() here, generate a bearer token and pass this both to the functions that build the payload that is sent to icasework, but also the super.prepare() step which calls the final pdf application, allowing us to update all upload urls with the token bearer at once.

P.s. I think this process is overly complicated as this took me ages to work out. I want to gradually refactor this into something far more sensible. This whole override BS is completely unnecessary imho and I reckon we can simplify it for others.